cmd/kuberhealthy/kuberhealthy.go

```diff
@@ -302,6 +235,7 @@ func (k *Kuberhealthy) watchForKHCheckChanges(ctx context.Context, c chan struct
 				c <- struct{}{}
 			case watch.Deleted:
 				log.Debugln("khcheck monitor saw a deleted event")
+				go k.deleteRelatedKhstate(*khc.Object.(*khcheckcrd.KuberhealthyCheck))
```

Comment:
Can you do your type assertion this on the prior line and check the error value for it while you're there?

https://golang.org/ref/spec#Type_assertions

Reply:
watch event
```
// Event represents a single event to a watched resource.
// +k8s:deepcopy-gen=true
type Event struct {
	Type EventType

	// Object is:
	//  * If Type is Added or Modified: the new state of the object.
	//  * If Type is Deleted: the state of the object immediately before deletion.
	//  * If Type is Bookmark: the object (instance of a type being watched) where
	//    only ResourceVersion field is set. On successful restart of watch from a
	//    bookmark resourceVersion, client is guaranteed to not get repeat event
	//    nor miss any events.
	//  * If Type is Error: *api.Status is recommended; other types may make sense
	//    depending on context.
	Object runtime.Object
}
```
It seems that only type is Error will return * api.Status, otherwise it will return the watch object itself

But adding a type check is also good, I will fix it

Reply:
Thank you!